aci/data_source_aci_igmpifp.go

```diff
@@ -0,0 +1,57 @@
+package aci
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ciscoecosystem/aci-go-client/v2/client"
+	"github.com/ciscoecosystem/aci-go-client/v2/models"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceAciIGMPInterfaceProfile() *schema.Resource {
+	return &schema.Resource{
+		ReadContext:   dataSourceAciIGMPInterfaceProfileRead,
+		SchemaVersion: 1,
+		Schema: AppendBaseAttrSchema(AppendNameAliasAttrSchema(map[string]*schema.Schema{
+			"logical_interface_profile_dn": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"relation_igmp_rs_if_pol": {
+				Type: schema.TypeString,
+
+				Optional:    true,
+				Description: "Query igmp:IfPol relationship object",
+			}})),
+	}
+}
+
+func dataSourceAciIGMPInterfaceProfileRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	aciClient := m.(*client.Client)
+	LogicalInterfaceProfileDn := d.Get("logical_interface_profile_dn").(string)
+	rn := fmt.Sprintf(models.RnIgmpIfP)
+	dn := fmt.Sprintf("%s/%s", LogicalInterfaceProfileDn, rn)
```

Comment:
we could remove the variable declare since it is not needed anywhere else?
```suggestion
	rn := fmt.Sprintf(models.RnIgmpIfP)
	dn := fmt.Sprintf("%s/%s", d.Get("logical_interface_profile_dn").(string), rn)
```

Reply:
same applies for other data-sources